futures-util/src/try_future/mod.rs

```diff
@@ -501,4 +504,136 @@ pub trait TryFutureExt: TryFuture {
     {
         IntoFuture::new(self)
     }
+
+    /// Joins the result of two futures, waiting for them both to complete or
+    /// for one to produce an error.
+    ///
+    /// This function will return a new future which awaits both this and the
+    /// `other` future to complete. If successful, the returned future will
+    /// finish with a tuple of both results. If unsuccesful, it will complete
+    /// with an error.
+    ///
+    /// Note that this function consumes the receiving future and returns a
+    /// wrapped version of it.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(async_await, await_macro, futures_api)]
+    /// # futures::executor::block_on(async {
+    /// use futures::future::{self, TryFutureExt};
+    ///
+    /// let a = future::ready(Ok::<i32, i32>(1));
+    /// let b = future::ready(Err::<i32, i32>(2));
+    /// let pair = a.try_join(b);
+    ///
+    /// assert_eq!(await!(pair), Err(2));
+    /// # });
```

Comment:
Missing error example

Reply:
I'm not sure what you're asking for here-- the docs show an example of an error occurring.

Reply:
Ah yes, you're right. It doesn't show the tuple case. I'd like two examples one with the success value tuple and one with the error

Reply:
Done.